backend/src/impl/init.py

```diff
@@ -0,0 +1,42 @@
+import logging
+import os
+from typing import Final
+
+from explainaboard_web.impl.config import (
+    LocalDevelopmentConfig,
+    ProductionConfig,
+    StagingConfig,
+)
+from flask import Flask
+
+
+def init(app: Flask) -> Flask:
+    """Initializes the flask app"""
+    _init_config(app)
+    _init_gcp_credentials()
+    return app
+
+
+def _init_config(app: Flask):
+    FLASK_ENV = os.getenv("FLASK_ENV")
+    if FLASK_ENV == "production":
+        app.config.from_object(ProductionConfig())
+    elif FLASK_ENV == "development":
+        app.config.from_object(LocalDevelopmentConfig())
+    elif FLASK_ENV == "staging":
+        app.config.from_object(StagingConfig())
+
+
+def _init_gcp_credentials():
```

Comment:
If the AWS secret manager does not provide the functionality to mount secrets as a file, it is better to directly create the credentials object using `google.auth` rather than creating additional file manually.

```python
# In Storage.__init__():
credentials = google.oauth2.service_account.from_service_account_info(json.loads(secret))
client = google.cloud.storage.Client(..., credentials=credentials)
```